packages/server/api/src/app/flows/flow.module.ts

```diff
@@ -18,14 +19,24 @@ export const flowModule: FastifyPluginAsyncTypebox = async (app) => {
     await app.register(testTriggerController, { prefix: '/v1/test-trigger' })
     websocketService.addListener(WebsocketServerEvent.TEST_FLOW_RUN, (socket) => {
         return async (data: TestFlowRunRequestBody) => {
+            const eventEmitter = new EventEmitter() 
             const principal = await accessTokenManager.extractPrincipal(socket.handshake.auth.token)
             const flowRun = await flowRunService.test({
                 projectId: principal.projectId,
                 flowVersionId: data.flowVersionId,
             })
+
             socket.emit(WebsocketClientEvent.TEST_FLOW_RUN_STARTED, flowRun)
-            await engineResponseWatcher.listen(flowRun.id, false)
-            socket.emit(WebsocketClientEvent.TEST_FLOW_RUN_FINISHED, flowRun)
+            
+            eventEmitter.on('FlowStatus', (flowRunResponse: FlowRun) => {
+                if (flowRunResponse.status === FlowRunStatus.SUCCEEDED || flowRunResponse.status === FlowRunStatus.FAILED) {
```

Comment:
Add function somewhere maybe in shared call it is it terminal state?

It also includes other types like quota exceeded an internal error